lib/calendar.ts

```diff
@@ -1613,39 +1536,42 @@ const helperIslamic: HelperSharedImpl = ObjectAssign({}, helperSharedImpl, {
   DAYS_PER_ISLAMIC_YEAR: 354 + 11 / 30,
   DAYS_PER_ISO_YEAR: 365.2425,
   constantEra: 'ah',
-  estimateIsoDate(this: typeof helperIslamic & HelperSharedImpl, calendarDate) {
+  estimateIsoDate(
+    this: HelperPerCalendarImpl & { DAYS_PER_ISLAMIC_YEAR: number; DAYS_PER_ISO_YEAR: number },
```

Comment:
Is it worth defining a type alias for this, or is it just going to go away anyway when we use ES6 classes here?